src/dir_walker.rs

```diff
@@ -167,6 +191,9 @@ fn walk(dir: PathBuf, walk_data: &WalkData, depth: usize) -> Option<Node> {
                                         data.is_file(),
                                         walk_data.by_filecount,
                                         depth,
+                                        filter_modified_time,
```

Comment:
Consider changing `build_node` to accept a `walk_data` argument instead of individually 7 of `walk_data`'s fields

Reply:
Indeed, you are right, I reorganized the code according to your ideas. I will be very happy if you can help me review the code again. :)